packages/dev/core/src/Materials/greasedLinePluginMaterial.ts

```diff
@@ -0,0 +1,668 @@
+                `,
+            fragment: `
+                uniform vec4 grl_dashOptions;
+                uniform vec4 grl_colorMode_visibility_colorsWidth_useColors;
+                uniform vec3 grl_singleColor;
+                `,
+        };
+    }
+
+    // only getter, it doesn't make sense to use this plugin on a mesh other than GreasedLineMesh
+    // and it doesn't make sense to disable it on the mesh
+    get isEnabled() {
+        return true;
+    }
+
+    /**
+     * Bind the uniform buffer
+     * @param uniformBuffer
+     */
+    bindForSubMesh(uniformBuffer: UniformBuffer) {
+        const activeCamera = this._scene.activeCamera;
+
+        if (activeCamera) {
+            const projection = activeCamera.getProjectionMatrix();
+            uniformBuffer.updateMatrix("grl_projection", projection);
+        } else {
+            throw Error("GreasedLinePluginMaterial requires an active camera.");
+        }
+
+        const resolutionLineWidth = TmpVectors.Vector3[0];
+        if (this._options.resolution) {
+            resolutionLineWidth.x = this._options.resolution.x;
+            resolutionLineWidth.y = this._options.resolution.y;
+        } else {
+            resolutionLineWidth.x = this._engine.getRenderWidth();
+            resolutionLineWidth.y = this._engine.getRenderHeight();
+        }
+        resolutionLineWidth.z = this._options.width ? this._options.width : this._options.sizeAttenuation ? 1 : 0.1;
+        uniformBuffer.updateVector3("grl_resolution_lineWidth", resolutionLineWidth);
+
+        const dashOptions = TmpVectors.Vector4[0];
+        dashOptions.x = GreasedLinePluginMaterial._BooleanToNumber(this._options.useDash);
+        dashOptions.y = this._dashArray ?? 0;
+        dashOptions.z = this._options.dashOffset ?? 0;
+        dashOptions.w = this._options.dashRatio ?? 0.5;
+        uniformBuffer.updateVector4("grl_dashOptions", dashOptions);
+
+        const colorModeVisibilityColorsWidthUseColors = TmpVectors.Vector4[1];
+        colorModeVisibilityColorsWidthUseColors.x = this._options.colorMode ?? GreasedLineMeshColorMode.COLOR_MODE_SET;
+        colorModeVisibilityColorsWidthUseColors.y = this._options.visibility ?? 1;
+        colorModeVisibilityColorsWidthUseColors.z = this._colorsTexture ? this._colorsTexture.getSize().width  : 0;
+        colorModeVisibilityColorsWidthUseColors.w = GreasedLinePluginMaterial._BooleanToNumber(this._options.useColors);
+        uniformBuffer.updateVector4("grl_colorMode_visibility_colorsWidth_useColors", colorModeVisibilityColorsWidthUseColors);
+
+        if (this._options.color) {
+            uniformBuffer.updateColor3("grl_singleColor", this._options.color ?? Color3.White());
+        }
+
+        if (this._colorsTexture) {
+            uniformBuffer.setTexture("grl_colors", this._colorsTexture);
+        } else if (this._engine.isWebGPU) {
+            uniformBuffer.setTexture("grl_colors", GreasedLinePluginMaterial._EmptyColorsTexture);
+        }
+

```

Comment:
This texture is needed only for WebGPU, because you always need to bind a texture once you define a sampler2D in your fragment shader even if you don't use it. This is not the case for WebGL. If you use the sampler2D in the shader there will be allways a texture set.

Reply:
> If you use the sampler2D in the shader there will be allways a texture set.

If you don't use the `sampler2D` but still have the declaration (as is the case), you'll get a random texture, which isn't very nice and sometimes makes debugging the shader code more difficult (this has happened to me several times). There's no harm with setting the texture in WebGL as well, so it's easier to do it indiscriminately (and it makes the code shorter!).